From 2aeb197765ee31f5550b7c80f340e46f3b829a76 Mon Sep 17 00:00:00 2001
From: nightshad <nightshad@e54a6e83-5905-42d5-8d5c-058d10e6a962>
Date: Wed, 23 Feb 2011 12:21:23 +0000
Subject: [PATCH] Try to fix issue #373 as reported on the bug tracker.

Reset controller before polling for halt. Somehow the
controller refuse to init after initializing from u-boot.



git-svn-id: http://www.coolstreamtech.de/coolstream_public_svn@1196 e54a6e83-5905-42d5-8d5c-058d10e6a962
---
 .../drivers/usb/host/ehci-cx2450x.c                |   16 ++++------------
 1 files changed, 4 insertions(+), 12 deletions(-)

diff --git a/THIRDPARTY/kernel/linux-2.6.26.8-cnxt/drivers/usb/host/ehci-cx2450x.c b/THIRDPARTY/kernel/linux-2.6.26.8-cnxt/drivers/usb/host/ehci-cx2450x.c
index 55eb8b1..ac30d09 100644
--- a/THIRDPARTY/kernel/linux-2.6.26.8-cnxt/drivers/usb/host/ehci-cx2450x.c
+++ b/THIRDPARTY/kernel/linux-2.6.26.8-cnxt/drivers/usb/host/ehci-cx2450x.c
@@ -83,7 +83,6 @@ int usb_ehci_cx2450x_probe(const struct hc_driver *driver, struct platform_devic
     }
 
     retval = usb_add_hcd(hcd, irq, IRQF_SHARED | IRQF_DISABLED);
-
     if (!retval) 
 	return retval;
 
@@ -143,8 +142,7 @@ static int ehci_cx2450x_setup(struct usb_hcd *hcd)
     ehci->caps = hcd->regs;
     ehci->regs = hcd->regs + HC_LENGTH(readl(&ehci->caps->hc_capbase));
 
-    /* Conexant USB controller has a TT. 
-       FIXME: Should we set high or lowspeed here ? */
+    /* Conexant USB controller has a TT. */
     hcd->has_tt = 1;
 
     dbg_hcs_params(ehci, "reset");
@@ -153,23 +151,16 @@ static int ehci_cx2450x_setup(struct usb_hcd *hcd)
     /* cache this readonly data; minimize chip reads */
     ehci->hcs_params = readl(&ehci->caps->hcs_params);
 
-    /* This forces the Conexant USB controller into HCD mode. It appears we crap
-       out if we don't do this before. Perhaps set this mode before we load the 
-       driver somewhere ? (ie bootloader which initializes basic hw. */
-    if(ehci_is_TDI(ehci))
-	tdi_reset(ehci);
-
+    ehci_reset(ehci);
     retval = ehci_halt(ehci);
     if (retval)
-    	return retval;
+	return retval;
 
     /* data structure init */
     retval = ehci_init(hcd);
     if (retval)
 	return retval;
 
-    ehci_reset(ehci);
-
 #if 0
     temp = HCS_N_CC(ehci->hcs_params) * HCS_N_PCC(ehci->hcs_params);
     temp &= 0x0F;
@@ -182,6 +173,7 @@ static int ehci_cx2450x_setup(struct usb_hcd *hcd)
 	       HCS_N_PORTS(ehci->hcs_params));
     }
 #endif
+
     retval = ehci_cx2450x_reinit(ehci);
     return retval;
 }
-- 
1.7.3.4

